### ISSUE TITLE
dev_loss for categorical SER appears to use only the last development sample

### ISSUE THREAD

**Issue by itoten** [Lab-MSP/MSP-Podcast_Challenge_IS2025#1]
Hi,

I noticed a possible issue in the validation loop of `train_eval_files/train_cat_ser.py` .

The code collects and concatenates predictions and labels from the entire development set. However, the dev loss is calculated using `emo_pred` and `y` .
```python
    total_pred = torch.cat(total_pred, 0)
    total_y = torch.cat(total_y, 0)
    loss = utils.CE_weight_category(emo_pred, y, class_weights_tensor)
```
Since the development batch size is 1, `emo_pred` and `y` appear to contain only the final development sample. Therefore, model selection may be based only on that sample.

Shouldn't the loss be calculated as follows?
```python
    loss = utils.CE_weight_category(total_pred, total_y, class_weights_tensor)
```
